src/openrct2/rct2/S6Exporter.cpp

```diff
@@ -560,8 +560,11 @@ void S6Exporter::ExportRide(rct2_ride* dst, const Ride* src)
     dst->turn_count_default = src->turn_count_default;
     dst->turn_count_banked = src->turn_count_banked;
     dst->turn_count_sloped = src->turn_count_sloped;
-    // Includes holes and (for some strange reason?!) sheltered_eights
-    dst->inversions = src->inversions;
+    if (dst->type == RIDE_TYPE_MINI_GOLF)
+        dst->inversions = std::min((uint8_t)src->holes, (uint8_t)RCT12_MAX_GOLF_HOLES);
+    else
+        dst->inversions = std::min((uint8_t)src->inversions, (uint8_t)RCT12_MAX_INVERSIONS);
```

Comment:
This won't give you nice results when you have more than 255 inversions / holes. You should cast after the min function has applied, let min operate on the larger sized integers.